app/services/analytics.ts

```diff
@@ -0,0 +1,79 @@
+import { action } from '@ember-decorators/object';
+import { service } from '@ember-decorators/service';
+import { get } from '@ember/object';
+import Service from '@ember/service';
+import Ember from 'ember';
+import config from 'ember-get-config';
+
+export default class Analytics extends Service {
+    @service metrics;
+    @service session;
+
+    @action
+    click(category, label, extraInfo) {
+        let extra = extraInfo;
+        if (extra && typeof extra !== 'string') {
+            // This is to remove the event object when used with onclick
+            extra = null;
+        }
+        get(this, 'metrics')
+            .trackEvent({
+                category,
+                action: 'click',
+                label,
+                extra,
+            });
+
+        return true;
+    }
+
+    track(category, actionName, label, extraInfo) {
+        let extra = extraInfo;
+        if (extra && typeof extra !== 'string') {
+            extra = null;
+        }
+        get(this, 'metrics')
+            .trackEvent({
+                category,
+                actionName,
+                label,
+                extra,
+            });
+        return true;
+    }
+
+    trackPage(
+        page: string,
+        title: string,
+        publicPrivate: 'public' | 'private' | 'n/a' = 'n/a',
```

Comment:
This would be an excellent case for an [enum](https://www.typescriptlang.org/docs/handbook/enums.html).

Reply:
Ah, I knew there was a better way!